http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -461,4 +544,124 @@ private void addScopes(String scopes, Set<String> allScopes) {
+        } else {
+            return signedRequest.getCompactSerialization();
+        }
+    }
+
+    private static KeyPair getkeyPair(OidcClientConfiguration deployment) throws IOException {
+        if (deployment.getRequestObjectSigningKeyStoreFile().contains(PROTOCOL_CLASSPATH)) {
+            deployment.setRequestObjectSigningKeyStoreFile(deployment.getRequestObjectSigningKeyStoreFile().replace(PROTOCOL_CLASSPATH, Objects.requireNonNull(Thread.currentThread().getContextClassLoader().getResource("")).getPath()));
+        }
+        if (!deployment.getRequestObjectSigningAlgorithm().equals(NONE) && deployment.getRequestObjectSigningKeyStoreFile() == null){
+            throw log.invalidKeyStoreConfiguration();
+        } else {
+            return JWTSigningUtils.loadKeyPairFromKeyStore(deployment.getRequestObjectSigningKeyStoreFile(),
+                    deployment.getRequestObjectSigningKeyStorePassword(), deployment.getRequestObjectSigningKeyPassword(),
+                    deployment.getRequestObjectSigningKeyAlias(), deployment.getRequestObjectSigningKeyStoreType());
+        }
+    }
+
+    public JsonWebSignature signRequest(JwtClaims jwtClaims) throws IOException, JoseException {
+        JsonWebSignature jsonWebSignature = new JsonWebSignature();
+        jsonWebSignature.setPayload(jwtClaims.toJson());
+
+        if (!deployment.getRequestObjectSigningAlgValuesSupported().contains(deployment.getRequestObjectSigningAlgorithm())) {
+            throw log.invalidRequestObjectSignatureAlgorithm();
+        } else {
+            if (deployment.getRequestObjectSigningAlgorithm().contains(NONE)) { //unsigned
+                jsonWebSignature.setAlgorithmConstraints(AlgorithmConstraints.NO_CONSTRAINTS);
+                jsonWebSignature.setAlgorithmHeaderValue(NONE);
+            } else if (deployment.getRequestObjectSigningAlgorithm().contains(HMAC_SHA256)
+                    || deployment.getRequestObjectSigningAlgorithm().contains(HMAC_SHA384)
+                    || deployment.getRequestObjectSigningAlgorithm().contains(HMAC_SHA512)) { //signed with symmetric key
+                jsonWebSignature.setAlgorithmHeaderValue(deployment.getRequestObjectSigningAlgorithm());
+                Key key = new HmacKey(deployment.getResourceCredentials().get("secret").toString().getBytes(StandardCharsets.UTF_8));   //the client secret is a shared secret between the server and the client
+                jsonWebSignature.setDoKeyValidation(false);     //skips validation so that size of the secret does not matter
+                jsonWebSignature.setKey(key);
+            } else { //signed with asymmetric key
+                KeyPair keyPair = getkeyPair(deployment);
+                jsonWebSignature.setKey(keyPair.getPrivate());
+                jsonWebSignature.setAlgorithmHeaderValue(deployment.getRequestObjectSigningAlgorithm());
+            }
+            jsonWebSignature.sign();
```

Comment:
When the signing algorithm is NONE, what does this return? Is the sign() call needed in this case?